.github/workflows/rails_tests.yml

```diff
@@ -117,36 +126,31 @@ jobs:
 
     steps:
       - name: Checkout
-        uses: actions/checkout@v3
+        uses: actions/checkout@v4
+
       - name: Set up dependencies
         run: |
-          apk add --no-cache \
-            nodejs yarn npm \
-            tzdata \
-            git \
-            bash \
-            freetds-dev \
-            icu icu-dev \
-            curl libcurl curl-dev \
-            imagemagick \
-            libmagic file-dev file \
-            build-base libxml2-dev libxslt-dev postgresql-dev \
-            libgcc libstdc++ libx11 glib libxrender libxext libintl ttf-dejavu ttf-droid ttf-freefont ttf-liberation ttf-freefont \
-            chromium nss freetype freetype-dev harfbuzz ca-certificates ttf-freefont \
-            lftp postgresql tmux postgis geos geos-dev \
-            shared-mime-info gpg gpg-agent
-
+          apk add --no-cache $(cat .github/dependencies.txt)
           echo "postgres:5432:*:postgres:postgres" > ~/.pgpass
           chmod 600 ~/.pgpass
 
-          yarn install --frozen-lockfile
-          gem install bundler --version=2.5.7
-
       - name: Install gems
         run: |
+          gem install bundler --version=2.5.7
           bundle config set --local without 'production staging development'
           bundle install --jobs 4 --retry 3
 
+      - name: Cache Node.js modules
+        uses: actions/cache@v4
```

Comment:
So this is introducing cacheing for the warehouse node modules, right? Is it related to the e2e tests somehow, or just incidental improvement?

Just wondering because I have had issues with the actions cache for npm modules in the past, so wondering if its actually entangled to this change or can be considered separately.

Reply:
There's no dependency on the e2e tests, we could take it out if it's too much. I was just taking the opportunity to improve performance since my head was in there.

This is also only changing the warehouse assets. It's not related to the front end

Reply:
Okay gotcha, thanks for clarifying. This seems like an improvement overall so I say lets keep it!  thanks